source/OutfitterPanel.cpp

```diff
@@ -694,7 +700,7 @@ void OutfitterPanel::Buy(bool toCargo) const
 	}
 
 	int modifier = Modifier();
-	for(int i = 0; i < modifier && (CanBuy() || (toCargo && CanBuyToCargo())); ++i)
+	for(int i = 0; i < modifier && ((toCargo && CanBuyToCargo()) || CanBuy()); ++i)
```

Comment:
I think this right here was the biggest impact on performance.  Rearranged to short circuit more cleanly.

Reply:
... or maybe still not.  I think it's better, but still not perfect.  Anyone have any ideas why it would work slowly the first time (the first time I buy 888k javelins) but quickly (as quickly as selling) the next time?

Reply:
Solved.  I let the depreciation algorithm figure out the pricing and did truly bulk buys.  See revised BuyIntoCargo.